show/muxcable.py

```diff
@@ -0,0 +1,296 @@
+        port_table_keys[asic_id] = y_cable_tbl[asic_id].getKeys()
+
+    if port is not None:
+        asic_index = platform_sfputil.get_asic_id_for_logical_port(port)
+        if asic_index is None:
+            click.echo("Got invalid asic index for port {}, cant retreive mux status".format(port))
+            return
+
+        y_cable_asic_table = y_cable_tbl.get(asic_index, None)
+        if y_cable_asic_table is not None:
+            y_cable_asic_table_keys = y_cable_asic_table.getKeys()
+            if port in y_cable_asic_table_keys:
+
+                if json_flag:
+                    port_status_dict = {}
+                    port_status_dict["mux_cable"] = {}
+
+                    status_value = get_value_for_key_in_tbl(y_cable_asic_table, port, "status")
+                    port_status_dict["mux_cable"][port] = {"status": status_value}
+
+                    click.echo("muxcable Ports status : \n{}".format(json.dumps(port_status_dict, indent=4)))
+                else:
+                    print_data = {}
+
+                    status_value = get_value_for_key_in_tbl(y_cable_asic_table, port, "status")
+                    print_data[port] = status_value
+                    headers = ['port', 'status']
+                    data = sorted([(k, v) for k, v in print_data.items()])
+
+                    click.echo(tabulate(data, headers=headers))
+            else:
+                click.echo("this is not a valid port present on mux_cable".format(port))
+        else:
+            click.echo("there is not a valid asic table for this asic_index".format(asic_index))
+
+    else:
+
+        if json_flag:
+            port_status_dict = {}
+            port_status_dict["mux_cable"] = {}
+            for namespace in namespaces:
+                asic_id = multi_asic.get_asic_index_from_namespace(namespace)
+                for port in port_table_keys[asic_id]:
+                    status_value = get_value_for_key_in_tbl(y_cable_tbl[asic_id], port, "status")
+                    port_status_dict["mux_cable"][port] = {"status": status_value}
+
+            click.echo("muxcable Ports status : \n{}".format(json.dumps(port_status_dict, indent=4)))
+        else:
+            print_data = {}
+            for namespace in namespaces:
+                asic_id = multi_asic.get_asic_index_from_namespace(namespace)
+                for port in port_table_keys[asic_id]:
+                    status_value = get_value_for_key_in_tbl(y_cable_tbl[asic_id], port, "status")
+                    print_data[port] = status_value
+
+            headers = ['port', 'status']
+            data = sorted([(k, v) for k, v in print_data.items()])
+            click.echo(tabulate(data, headers=headers))
+
+
+@muxcable.command()
+@click.argument('port', required=False, default=None)
+@click.option('--json', 'json_flag', required=False, is_flag=True, type=click.BOOL, help="display the output in json format")
```

Comment:
Suggest renaming `json_flag` to `json_output`

Reply:
fixed

Reply:
Doesn't seem to be fixed.

Reply:
fixed